lib/node_modules/@stdlib/lapack/base/dgttrf/lib/base.js

```diff
@@ -0,0 +1,172 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var abs = require( '@stdlib/math/base/special/abs' );
+
+
+// MAIN //
+
+/**
+* Computes an `LU` factorization of a real tri diagonal matrix `A` using elimination with partial pivoting and row interchanges.
+*
+* ## Notes
+*
+* -   On exit, `DL` is overwritten by the multipliers that define the matrix `L` from the `LU` factorization of `A`.
+* -   On exit, `D` is overwritten by the diagonal elements of the upper triangular matrix `U` from the `LU` factorization of `A`.
+* -   On exit, `DU` is overwritten by the elements of the first super-diagonal of `U`.
+* -   On exit, `DU2` is overwritten by the elements of the second super-diagonal of `U`.
+* -   On exit, for 0 <= i < n, row i of the matrix was interchanged with row IPIV(i).  IPIV(i) will always be either i or i+1; IPIV(i) = i indicates a row interchange was not required.
+*
+* @private
+* @param {NonNegativeInteger} N - order of matrix A.
+* @param {Float64Array} DL - sub diagonal elements of A.
+* @param {integer} sdl - stride length for `DL`
+* @param {NonNegativeInteger} odl - starting index of `DL`
+* @param {Float64Array} D - diagonal elements of A.
+* @param {integer} sd - stride length for `D`
+* @param {NonNegativeInteger} od - starting index of `D`
+* @param {Float64Array} DU - super diagonal elements of A.
+* @param {integer} sdu - stride length for `DU`
+* @param {NonNegativeInteger} odu - starting index of `DU`
+* @param {Float64Array} DU2 - vector to store the second super diagonal of `U`
+* @param {integer} sdu2 - stride length for `DU2`
+* @param {NonNegativeInteger} odu2 - starting index of `DU2`
+* @param {Int32Array} IPIV - vector of pivot indices
+* @param {integer} si - stride length for `IPIV`
+* @param {NonNegativeInteger} oi - starting index of `IPIV`
+* @returns {integer} status code
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+* var Int32Array = require( '@stdlib/array/int32' )
+*
+* var DL = new Float64Array( [ 1.0, 1.0 ] );
+* var D = new Float64Array( [ 2.0, 3.0, 1.0 ] );
+* var DU = new Float64Array( [ 1.0, 1.0 ] );
+* var DU2 = new Float64Array( 1 );
+* var IPIV = new Int32Array( 3 );
+*
+* dgttrf( 3, DL, 1, 0, D, 1, 0, DU, 1, 0, DU2, 1, 0, IPIV, 1, 0 );
+* // DL => <Float64Array>[ 0.5, 0.4 ]
+* // D => <Float64Array>[ 2, 2.5, 0.6 ]
+* // DU => <Float64Array>[ 1, 1 ]
+* // DU2 => <Float64Array>[ 0 ]
+* // IPIV => <Int32Array>[ 0, 1, 2 ]
+*/
+function dgttrf( N, DL, sdl, odl, D, sd, od, DU, sdu, odu, DU2, sdu2, odu2, IPIV, si, oi ) { // eslint-disable-line max-len, max-params
+	var fact;
+	var temp;
+	var idu2;
+	var idu;
+	var idl;
+	var dli;
+	var dui;
+	var id;
+	var ip;
+	var di;
+	var ii;
+	var i;
+
+	// Quick return if possible
+	if ( N === 0 ) {
+		return 0;
+	}
+
+	idl = odl;
+	id = od;
+	idu = odu;
+	idu2 = odu2;
+	ip = oi;
+
+	// Initialise ith element of IPIV as i
```

Comment:
```suggestion
	// Initialize ith element of IPIV as i
```

American spellings here and elsewhere